code/go/0chain.net/smartcontract/common/wallet.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"encoding/hex"
+
+	"0chain.net/core/common"
+	"0chain.net/core/encryption"
+	"github.com/0chain/common/core/logging"
+)
+
+
+func ValidateWallet(publicKey, delegateWalletID string) *common.Error {
+	publicKeyBytes, err := hex.DecodeString(publicKey)
+	if err != nil {
+		return common.NewError("add_sharder",
+			"couldn't decode publick key to compare to delegate wallet")
+	}
+	operationalClientID := encryption.Hash(publicKeyBytes)
+
+	if operationalClientID == delegateWalletID {
+		logging.Logger.Error("couldn't use the same wallet as both operational and delegate")
+		return common.NewError("add_sharder",
+			"couldn't use the same wallet as both operational and delegate")
+	}
+
+	return nil
```

Comment:
Oh, actually this functions will be reused somewhere else and should be put in the `encryption` package, where we do the public/private/client ID things. I've done that in another [PR](https://github.com/0chain/0chain/pull/1850/files#diff-a4b5842743d9fb9c0c1e8f9d21e4b238ccf11fa0cb59fef25a8890ce57180622R176-R188) but not merged it, let me check if we can merge it today.